arches_querysets/rest_framework/serializers.py

```diff
@@ -560,16 +562,21 @@ def __init__(self, instance=None, data=empty, *, context=None, **kwargs):
         self._child_nodegroup_aliases = []
 
     def to_representation(self, data):
-        """Prevent newly minted blank tiles from serializing with pk's."""
+        """Prevent newly minted blank tiles from serializing with pk's if the
+        fill_blanks option is in force."""
         ret = super().to_representation(data)
-        if isinstance(data, TileTree):
+        if self.context["fill_blanks"] and isinstance(data, TileTree):
```

Comment:
@apeters that's interesting that the tileid serialization on create already works on main, must have been fixed as a side effect somewhere else. That just leaves this change here as "adding safety", so that code that is only supposed to run for the `?fill_blanks=true` option definitely only runs when that option is in force.

Then the other things going on this PR:
- more tests
- PR #115 got merged into this one instead of main to prevent merge conflicts. It fixes some edit log and transaction id behavior by moving code that shouldn't have been in the DRF layer out of it.

My advice is to merge it -- I can respond if there are unexpected issues with it.

Reply:
One nice thing that got fixed in #115 is that you can actually use the browsable API to just save tiles back immediately by scrolling down and clicking POST, leaving all the default values in place.

When I try that on main, I get TypeError, but it's fixed & tested here. I think @aarongundel was saying something about "I should be able save default values back" not working, not sure if that's the same failure fixed here.